webknossos/webknossos/client/_upload_dataset.py

```diff
@@ -25,55 +24,74 @@ def _cached_get_upload_datastore(context: _WebknossosContext) -> str:
     raise ValueError("No datastore found where datasets can be uploaded.")
 
 
+def _walk_file(path: Path, base_path: Path) -> Iterator[Tuple[Path, Path, int]]:
+    if path.is_dir():
+        yield from _walk(path, base_path)
+        return
+    elif path.is_symlink():
+        yield from _walk_file(path.resolve(), base_path)
+        return
+    yield (path.resolve(), path.relative_to(base_path), path.stat().st_size)
+
+
+def _walk(
+    path: Path, base_path: Optional[Path] = None
+) -> Iterator[Tuple[Path, Path, int]]:
+    if base_path is None:
+        base_path = path
+    for p in Path(path).iterdir():
+        yield from _walk_file(p, base_path)
```

Comment:
I think this can be simplified quite a bit, removing `_walk_file`:
```suggestion
def _walk(
    path: Path, base_path: Optional[Path] = None
) -> Iterator[Tuple[Path, Path, int]]:
    if base_path is None:
        base_path = path
    if path.is_dir():
        for p in Path(path).iterdir():
            yield from _walk(p, base_path)
    else:
        yield (path.resolve(), path.relative_to(base_path), path.stat().st_size)
```

I think it should be better to only resolve a path for the yield, not before in the symlink case, since the `relative_to` might yield something outside of the actual directory we're operating in.

Reply:
Very good. I wrongly thought that `p.is_dir() == False` for symlink-dirs.